### ISSUE TITLE
Server side fetch() fails for certain redirects

### ISSUE THREAD

**Issue by seriousm4x** [sveltejs/kit#7834]
### Describe the bug

Running a server side `fetch()` for an api that is hosted behind a cloudflare proxy fails. Fetching other apis works.

### Reproduction

Basically in your `+page.server.js` run:
```js
export async function load() {
    const URL = "https://some-api-behind-cloudflare";
    const req = await fetch(URL);
    const data = await req.json();

    console.log(data);

    return {
        data: data
    }
}
```

and in `+page.svelte`:
```js
<script>
    export let data;
</script>

{JSON.stringify(data)}
```

I've created a barebones repo to easily reproduce this over at https://github.com/seriousm4x/sveltekit-ssr-fetch-cloudflare-proxy
Under `src/routes/fetch/+page.server.js` are 2 URLs. `api.wubbl0rz.tv` is my own api running behind cloudflare proxy, `catfact.ninja` is a random free api I found in the internet. Comment/Uncomment them and open your brower at [http://localhost:5173/fetch](http://localhost:5173/fetch) to see the error. 

### Logs

```Shell
TypeError: fetch failed
    at fetch (/home/max/repos/sveltekit-ssr-fetch-cloudflare-proxy/node_modules/.pnpm/undici@5.12.0/node_modules/undici/index.js:105:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async load (/src/routes/fetch/+page.server.js:5:17)
    at async load_server_data (file:///home/max/repos/sveltekit-ssr-fetch-cloudflare-proxy/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.563_svelte@3.53.1+vite@3.2.4/node_modules/@sveltejs/kit/src/runtime/server/page/load_data.js:33:17)
    at async file:///home/max/repos/sveltekit-ssr-fetch-cloudflare-proxy/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.563_svelte@3.53.1+vite@3.2.4/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:142:13
```

### System Info

```Shell
System:
    OS: Linux 6.0 undefined
    CPU: (32) x64 AMD Ryzen 9 5950X 16-Core Processor
    Memory: 57.36 GB / 62.71 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 19.1.0 - /usr/bin/node
    npm: 8.19.2 - /usr/bin/npm
  Browsers:
    Chromium: 107.0.5304.110
    Firefox: 107.0
  npmPackages:
    @sveltejs/adapter-node: next => 1.0.0-next.101 
    @sveltejs/kit: next => 1.0.0-next.563 
    svelte: ^3.53.1 => 3.53.1 
    vite: ^3.2.4 => 3.2.4
```

### Severity

serious, but I can work around it

**Comment by Conduitry** [comment]
The issue here seems to be unrelated to Cloudflare, but rather that the redirect is not being followed correctly. https://api.wubbl0rz.tv/vods redirects to https://api.wubbl0rz.tv/vods/ and if I directly request `/vods/` instead, it works.

I'm not sure what's causing undici to fail on making this request. It also happens if I run `fetch('https://api.wubbl0rz.tv/vods')` in the REPL. Perhaps it has to do with that response being a 301 with an HTML body.

**Comment by seriousm4x** [comment]
Huh, that actually seems to be the issue. I'm really confused though, because I have a clone of that api for development on my local net where it just works fine. So I was certain that cloudflare was the issue.

**Comment by repsac-by** [comment]
Fixed by #7995
